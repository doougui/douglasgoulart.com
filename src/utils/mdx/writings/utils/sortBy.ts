import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
