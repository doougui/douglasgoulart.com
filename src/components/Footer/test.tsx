import { render } from 'utils/tests';

import { Footer } from '.';

describe('<Footer />', () => {
  it('should render correctly', () => {
    const { container } = render(<Footer />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
