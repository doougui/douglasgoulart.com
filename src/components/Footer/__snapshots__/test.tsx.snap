// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render correctly 1`] = `
.c12 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-size: 1.6rem;
  line-height: 2.4rem;
  color: #EDEDEF;
  background-color: #107BEB;
  border-radius: 0.4rem;
  padding: 1rem 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c10 {
  all: unset;
  width: 100%;
  font-size: 1.6rem;
  background: #2D2D2F;
  padding: 1.2rem 1.5rem;
  border-radius: 0.4rem;
  color: #EDEDEF;
}

.c10::-webkit-input-placeholder {
  color: #888888;
}

.c10::-moz-placeholder {
  color: #888888;
}

.c10:-ms-input-placeholder {
  color: #888888;
}

.c10::placeholder {
  color: #888888;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 1rem;
}

.c9 button {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.c11 {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
  min-width: 10rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1.5rem;
  color: #EDEDEF;
  font-size: 1.6rem;
}

.c2 {
  font-family: Inter,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-weight: 600;
}

.c0 {
  margin-top: 5rem;
  padding-top: 3rem;
  border-top: 1px solid #2D2D2F;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  gap: 8rem;
  margin-top: 2.5rem;
}

.c4 {
  color: #EDEDEF;
  font-size: 1.6rem;
  line-height: 2.4rem;
  font-weight: 400;
}

.c8 {
  font-size: 1.6rem;
  color: #888888;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.5rem;
  margin-top: 1.5rem;
}

.c6 {
  color: #888888;
  font-size: 1.6rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c13 {
  color: #EDEDEF;
  font-size: 1.6rem;
  line-height: 2.4rem;
  margin-top: 3rem;
}

.c7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<footer
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      class=""
      color="#EDEDEF"
      fill="currentColor"
      height="26"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(237, 237, 239);"
      viewBox="0 0 24 24"
      width="26"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.55 2 12 2zm3.75 14.65c-2.35-1.45-5.3-1.75-8.8-.95-.35.1-.65-.15-.75-.45-.1-.35.15-.65.45-.75 3.8-.85 7.1-.5 9.7 1.1.35.15.4.55.25.85-.2.3-.55.4-.85.2zm1-2.7c-2.7-1.65-6.8-2.15-9.95-1.15-.4.1-.85-.1-.95-.5-.1-.4.1-.85.5-.95 3.65-1.1 8.15-.55 11.25 1.35.3.15.45.65.2 1s-.7.5-1.05.25zM6.3 9.75c-.5.15-1-.15-1.15-.6-.15-.5.15-1 .6-1.15 3.55-1.05 9.4-.85 13.1 1.35.45.25.6.85.35 1.3-.25.35-.85.5-1.3.25C14.7 9 9.35 8.8 6.3 9.75z"
          fill-rule="nonzero"
        />
      </g>
    </svg>
    <span>
      Spotify: 
      <span
        class="c2"
      >
        Not Playing
      </span>
    </span>
  </div>
  <menu
    class="c3"
  >
    <section>
      <h3
        class="c4"
      >
        Navigation
      </h3>
      <nav
        class="c5"
      >
        <a
          class="c6"
          href="/"
        >
          Home
        </a>
        <a
          class="c6"
          href="/about"
        >
          About
        </a>
        <a
          class="c6"
          href="/writings"
        >
          Writings
        </a>
        <a
          class="c6"
          href="/guestbook"
        >
          Guestbook
        </a>
      </nav>
    </section>
    <section>
      <h3
        class="c4"
      >
        Social
      </h3>
      <nav
        class="c5"
      >
        <a
          class="c6"
          href="https://github.com/doougui"
          target="_blank"
        >
          GitHub
        </a>
        <a
          class="c6"
          href="https://www.linkedin.com/in/douglaspigoulart/"
          target="_blank"
        >
          Linkedin
        </a>
        <a
          class="c6"
          href="https://twitter.com/oDougui"
          target="_blank"
        >
          Twitter
        </a>
        <a
          class="c6"
          href="mailto:douglaspigoulart@gmail.com"
          target="_blank"
        >
          E-mail
        </a>
      </nav>
    </section>
    <section>
      <h3
        class="c4"
      >
        Extras
      </h3>
      <nav
        class="c5"
      >
        <a
          class="c6"
          href="/"
          target="_blank"
        >
          Newsletter
        </a>
        <a
          class="c6"
          href="/"
          target="_blank"
        >
          Resume
        </a>
        <a
          class="c6"
          href="https://gitlab.com/douglaspigoulart"
          target="_blank"
        >
          GitLab
        </a>
      </nav>
    </section>
    <section
      class="c7"
    >
      <h3
        class="c4"
      >
        Newsletter
      </h3>
      <p
        class="c8"
      >
        Get new writings delivered to your inbox!
      </p>
      <form>
        <div
          class="c9"
        >
          <input
            class="c10 c11"
            name="email"
            placeholder="johndoe@email.com"
            value=""
          />
          <button
            class="c12"
            type="submit"
          >
            Subscribe
          </button>
        </div>
      </form>
    </section>
  </menu>
  <p
    class="c13"
  >
    Â© 2022 Douglas Pinheiro Goulart
  </p>
</footer>
`;
